jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -91,4 +98,24 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             return Optional.of(header.equals("1"));
         }
     }
+
+    private static void addUserIdMetadataToSpan(ContainerRequestContext requestContext) {
+        Optional<String> userId = getUserIdFromAuthHeader(requestContext);
+        if (userId.isPresent()) {
+            Tracer.addSpanMetadata(ImmutableMap.of(USER_ID_METADATA_KEY, userId.get()));
+        }
+    }
+
+    private static Optional<String> getUserIdFromAuthHeader(ContainerRequestContext requestContext) {
+        String authHeaderString = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
+        if (authHeaderString == null) {
+            return Optional.empty();
+        } else {
+            AuthHeader authHeader = AuthHeader.valueOf(authHeaderString);
+            BearerToken bearerToken = authHeader.getBearerToken();
+            UnverifiedJsonWebToken token = UnverifiedJsonWebToken.of(bearerToken);
```

Comment:
What happens if this method throws due to invalid bearer token? I assume we should `return Optional.empty()`. Can we add tests that cover this case?